kolibri/utils/file_transfer.py

```diff
@@ -75,113 +84,390 @@ def retry_import(e):
     return False
 
 
-class Transfer(object):
+# Set block size to 128KB
+# the previous value of 2MB was set to avoid frequent progress
+# updates during file transfer, but since file transfers
+# have been parallelized, and individual file downloads are not tracked
+# except as part of overall download progress, this is no longer necessary.
+# 128KB allows for small chunks of files to be transferred
+# with the potential for interruption, while still allowing
+# for a reasonable amount of data to be transferred in one go.
+# This will also reduce memory usage when transferring large files.
+BLOCK_SIZE = 128 * 1024
+
+
+class ChunkedFile(BufferedIOBase):
+    def __init__(self, filepath):
+        self.filepath = filepath
+        self.chunk_dir = filepath + ".chunks"
+        mkdirp(self.chunk_dir, exist_ok=True)
+        self.chunk_size = BLOCK_SIZE
+        self.position = 0
+        self._file_size = None
+
+    @property
+    def chunks_count(self):
+        return int(math.ceil(float(self.file_size) / float(self.chunk_size)))
+
+    @property
+    def file_size(self):
+        if self._file_size is not None:
+            return self._file_size
+        try:
+            with open(os.path.join(self.chunk_dir, ".file_size"), "r") as f:
+                self._file_size = int(f.read())
+        except (OSError, IOError, ValueError):
+            raise ValueError("file_size is not set")
+        return self._file_size
+
+    @file_size.setter
+    def file_size(self, value):
+        with open(os.path.join(self.chunk_dir, ".file_size"), "w") as f:
```

Comment:
Very minor thing, but seems unnecessary to prepend `.` to `file_size` and `chunk_*`. Any reason for it?

Reply:
No - we can remove. We could also move the file_size into the cache that I use for the file download locking too, so there are definitely some improvements to be made.